database-commons/src/main/java/io/cdap/plugin/db/connector/AbstractDBSpecificConnector.java

```diff
@@ -117,16 +118,24 @@ protected String getConnectionString(String database) {
     return config.getConnectionString();
   }
 
+  protected String getTableName(String database, String schema, String table) {
+    return schema == null ? String.format("\"%s\".\"%s\"", database, table)
```

Comment:
Hm. I see that it's preexisting code, but which DB is it targeted for? Usually A.B. is schema.table, not database.table

Reply:
Actually it's better to make it abstract. As far as I can see all subclasses override it. And `SqlServerConnector` override just one method, so it's even more weird

Reply:
And I don't think database.table would work for SqlServer at all. https://docs.microsoft.com/en-us/sql/t-sql/queries/from-transact-sql?view=sql-server-ver16 gives `[ database_name . [ schema_name ] . | schema_name . ] table_or_view_name [ AS ] table_or_view_alias` format, so if one want to go schemaless it should be `database..table`, not `database.table`

Reply:
Hi Vitalli, thanks for the feedback!

These changes were part of the `getTableName()` refactor, and thus tried to preserve the previously implemented behavior as much as possible. The existence of this part of the code was kept because, while I could not find any subclasses of AbstractDBConnector.java in this repository, I was not sure if there were some elsewhere that depended on this part's existence (as I assumed it was originally implemented for a reason). If you believe it is fine to make this abstract, I can make that change.

As for SqlServer, I have not looked into whether this format works -- if not, then it was likely also broken previously. We can fix it, if so.

Reply:
I have just fixed SqlServer's table name to have the same behavior as previously. I am leaving `AbstractDBSpecificConnector`'s `getTableName()` function as is, for now, for the reasons listed.